src/alert-manager/src/cluster-utilization/send_alert.py

```diff
@@ -0,0 +1,194 @@
+from datetime import timezone, datetime
+import logging
+import os
+import urllib
+import requests
+
+CLUSTER_QUERY_STRING = "avg(avg_over_time(nvidiasmi_utilization_gpu[7d]))"
+JOB_QUERY_STRING = 'avg by (job_name) (avg_over_time(task_gpu_percent{username!="noves"}[7d]))'
+# user used gpu hours / total gpu hours
+USER_QUERY_STRING = \
+    "(sum by (username) (sum_over_time(task_gpu_percent[7d]))) / (sum by (username) (count_over_time(task_gpu_percent[7d])*100)) * 100"
+
+QUERY_PREFIX = "/prometheus/api/v1/query"
+ALERT_PREFIX = "/alert-manager/api/v1/alerts"
+REST_JOB_API_PREFIX = "/rest-server/api/v2/jobs"
+
+TOKEN = os.environ.get('PAI_BEARER_TOKEN')
+
+def enable_request_debug_log(func):
+    def wrapper(*args, **kwargs):
+        requests_log = logging.getLogger("urllib3")
+        level = requests_log.level
+        requests_log.setLevel(logging.DEBUG)
+        requests_log.propagate = True
+
+        try:
+            return func(*args, **kwargs)
+        finally:
+            requests_log.setLevel(level)
+            requests_log.propagate = False
+
+    return wrapper
+
+
+@enable_request_debug_log
+def get_job_usage_info(job_usage_result, rest_url):
+    job_infos = {}
+    job_usage = []
+    job_list = []
+    # get all jobs
+    headers = {'Authorization': "Bearer {}".format(TOKEN)}
+    resp = requests.get(rest_url, headers=headers)
+    if resp.ok:
+        job_list = resp.json()
+    else:
+        logging.warning("Failed to get job list")
+
+    for v in job_usage_result["data"]["result"]:
+        job_infos[v["metric"]["job_name"]] = {"usage": v["value"][1][:6] + "%"}
+    for job_name, job_info in job_infos.items():
+        url = urllib.parse.urljoin(rest_url + "/", job_name)
+        resp = requests.get(url, headers=headers)
+        if not resp.ok:
+            logging.warning("request failed %s", resp.text)
+            continue
+        resp_json = resp.json()
+        if not resp_json["jobStatus"]["appLaunchedTime"]:
+            logging.warning("job not start, ignore it")
+            continue
+        job_info["start_time"] = datetime.fromtimestamp(
+            int(resp_json["jobStatus"]["appLaunchedTime"]) / 1000,
+            timezone.utc)
+        if not resp_json["jobStatus"]["appLaunchedTime"] or not resp_json[
```

Comment:
Do not need to check `resp_json["jobStatus"]["appLaunchedTime"]` here?

Reply:
> Do not need to check `resp_json["jobStatus"]["appLaunchedTime"]` here?

Removed, thx